### ISSUE TITLE
Add "new version available" feature to flutter pub

### ISSUE THREAD

**Issue by dvorapa** [flutter/flutter#46226]
## Use case
Dart has got this awesome feature of showing, whenever there is a newer version of a dependency available, but pubspec.yaml asks for older one. This is crucial in development as you usually want to know, if new version of a dependency is out in order to update your package/app as well. This way you can maintain your package/app up-to-date, stable and secure.

## Proposal
To `flutter pub get` and `flutter pub upgrade` add a parameter turning this feature on or make it available by default.

OR

Add a new `flutter pub` command to check the up-to-dateness of the dependencies.

Comparison:
**Dart**
```
$ pub get
Resolving dependencies... (9.9s)
+ analyzer 0.32.6 (0.39.1 available)
+ archive 2.0.11
+ args 1.5.2
+ async 2.4.0
+ bazel_worker 0.1.23
+ build 0.12.8 (1.2.2 available)
+ build_config 0.3.2 (0.4.1+1 available)
+ build_modules 1.0.3 (2.7.0 available)
+ build_resolvers 0.2.2+7 (1.3.0 available)
+ build_runner 0.10.3 (1.7.2 available)
+ build_runner_core 1.0.0 (4.2.0 available)
+ build_web_compilers 0.4.4+3 (2.7.1 available)
+ built_collection 4.3.0
+ built_value 7.0.0
+ charcode 1.1.2
+ code_builder 3.2.1
+ collection 1.14.12
+ convert 2.1.1
+ crypto 2.1.3
+ csslib 0.16.1
+ dart_style 1.1.3 (1.3.3 available)
+ fixnum 0.10.11
+ front_end 0.1.4+2 (0.1.29 available)
+ glob 1.2.0
+ graphs 0.1.3+1 (0.2.0 available)
+ html 0.14.0+3
+ http_multi_server 2.1.0
+ http_parser 3.1.3
+ io 0.3.3
+ js 0.6.1+1
+ json_annotation 1.2.0 (3.0.0 available)
+ kernel 0.3.4+2 (0.3.29 available)
+ logging 0.11.3+2
+ matcher 0.12.6
+ meta 1.1.8
+ mime 0.9.6+3
+ node_interop 1.0.3
+ node_io 1.0.1+2
+ package_config 1.1.0
+ path 1.6.4
+ pedantic 1.9.0
+ plugin 0.2.0+3
+ pool 1.4.0
+ protobuf 1.0.1
+ pub_semver 1.4.2
+ pubspec_parse 0.1.4 (0.1.5 available)
+ quiver 2.1.2+1
+ scratch_space 0.0.4+1
+ shelf 0.7.5
+ shelf_web_socket 0.2.3
+ source_maps 0.10.8
+ source_span 1.5.5
+ stack_trace 1.9.3
+ stream_channel 2.0.0
+ stream_transform 0.0.20 (1.1.0 available)
+ string_scanner 1.0.5
+ term_glyph 1.1.0
+ timing 0.1.1+2
+ typed_data 1.1.6
+ watcher 0.9.7+13
+ web_socket_channel 1.1.0
+ yaml 2.2.0
Changed 62 dependencies!
Precompiling executables... (38.1s)
Precompiled build_runner:graph_inspector.
Precompiled build_runner:build_runner.
$ pub upgrade
Resolving dependencies... (10.4s)
  analyzer 0.32.6 (0.39.1 available)
  archive 2.0.11
  args 1.5.2
  async 2.4.0
  bazel_worker 0.1.23
  build 0.12.8 (1.2.2 available)
  build_config 0.3.2 (0.4.1+1 available)
  build_modules 1.0.3 (2.7.0 available)
  build_resolvers 0.2.2+7 (1.3.0 available)
  build_runner 0.10.3 (1.7.2 available)
  build_runner_core 1.0.0 (4.2.0 available)
  build_web_compilers 0.4.4+3 (2.7.1 available)
  built_collection 4.3.0
  built_value 7.0.0
  charcode 1.1.2
  code_builder 3.2.1
  collection 1.14.12
  convert 2.1.1
  crypto 2.1.3
  csslib 0.16.1
  dart_style 1.1.3 (1.3.3 available)
  fixnum 0.10.11
  front_end 0.1.4+2 (0.1.29 available)
  glob 1.2.0
  graphs 0.1.3+1 (0.2.0 available)
  html 0.14.0+3
  http_multi_server 2.1.0
  http_parser 3.1.3
  io 0.3.3
  js 0.6.1+1
  json_annotation 1.2.0 (3.0.0 available)
  kernel 0.3.4+2 (0.3.29 available)
  logging 0.11.3+2
  matcher 0.12.6
  meta 1.1.8
  mime 0.9.6+3
  node_interop 1.0.3
  node_io 1.0.1+2
  package_config 1.1.0
  path 1.6.4
  pedantic 1.9.0
  plugin 0.2.0+3
  pool 1.4.0
  protobuf 1.0.1
  pub_semver 1.4.2
  pubspec_parse 0.1.4 (0.1.5 available)
  quiver 2.1.2+1
  scratch_space 0.0.4+1
  shelf 0.7.5
  shelf_web_socket 0.2.3
  source_maps 0.10.8
  source_span 1.5.5
  stack_trace 1.9.3
  stream_channel 2.0.0
  stream_transform 0.0.20 (1.1.0 available)
  string_scanner 1.0.5
  term_glyph 1.1.0
  timing 0.1.1+2
  typed_data 1.1.6
  watcher 0.9.7+13
  web_socket_channel 1.1.0
  yaml 2.2.0
No dependencies changed.
```

**Flutter**
```
$ flutter pub get
Running "flutter pub get" in myapp...                            2,9s
$ flutter pub upgrade
Running "flutter pub upgrade" in myapp...                        8,8s
```

**Comment by dvorapa** [comment]
Seems solved by Flutter 2.0.0

**Comment by christopherfujino** [comment]
`flutter pub outdated` now works
